Content.Server/Paper/PaperSystem.cs

```diff
@@ -1,20 +1,139 @@
 using Content.Server.UserInterface;
 using Content.Shared.Paper;
+using Content.Shared.Examine;
+using Content.Shared.Interaction;
+using Content.Shared.Tag;
+using Robust.Server.GameObjects;
+using Content.Server.Popups;
+using Robust.Shared.Player;
+
+using static Content.Shared.Paper.SharedPaperComponent;
 
 namespace Content.Server.Paper
 {
     public sealed class PaperSystem : EntitySystem
     {
+        [Dependency] private readonly TagSystem _tagSystem = default!;
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
         public override void Initialize()
         {
             base.Initialize();
+
+            SubscribeLocalEvent<PaperComponent, ComponentInit>(OnInit);
             SubscribeLocalEvent<PaperComponent, BeforeActivatableUIOpenEvent>(AfterUIOpen);
+            SubscribeLocalEvent<PaperComponent, ExaminedEvent>(OnExamined);
+            SubscribeLocalEvent<PaperComponent, InteractUsingEvent>(OnInteractUsing);
+            SubscribeLocalEvent<PaperComponent, PaperInputTextMessage>(OnInputTextMessage);
+        }
+
+        private void OnInit(EntityUid uid, PaperComponent paperComp, ComponentInit args)
+        {
+            paperComp.Mode = PaperAction.Read;
+            UpdateUserInterface(uid, paperComp);
+        }
+
+        private void AfterUIOpen(EntityUid uid, PaperComponent paperComp, BeforeActivatableUIOpenEvent args)
```

Comment:
why is it called `AfterUIOpen` if this is the before event

Reply:
Damn, I read right over that one, it was the only method that was in paper system before, changing the name.